libclc/amdgcn-amdhsa/libspirv/math/log2.cl

```diff
@@ -0,0 +1,19 @@
+//===----------------------------------------------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include <clcmacro.h>
+#include <spirv/spirv.h>
+ 
+double __ocml_log2_f64(double);
+float __ocml_log2_f32(float);
+ 
+#define __CLC_FUNCTION __spirv_ocl_log2
+#define __CLC_BUILTIN __ocml_log2
+#define __CLC_BUILTIN_F __CLC_XCONCAT(__CLC_BUILTIN, _f32)
+#define __CLC_BUILTIN_D __CLC_XCONCAT(__CLC_BUILTIN, _f64)
+#include <math/unary_builtin.inc>
```

Comment:
How about defining `__CLC_BUILTIN_H` for `log2` and `remainder` to make them consistent with `ilogb`, which is defined for `half` data type?

Reply:
Change made. Just defaulting to `__CLC_BUILTIN_F`

Reply:
Sorry, I missed that <math/unary_builtin.inc> defines function overload for half data type and uses float built-in by default to implement it.

Reply:
Sorry I missed that too. Will revert last commit